applications/FluidDynamicsApplication/custom_elements/vectorial_convection_fractional_element.cpp

```diff
@@ -0,0 +1,725 @@
+const double crhs23 = crhs20 + crhs21 + crhs22;
+const double crhs24 = N[0]*(bdf0*vfrac(0,1) + bdf1*vn(0,1) + bdf2*vnn(0,1)) + N[1]*(bdf0*vfrac(1,1) + bdf1*vn(1,1) + bdf2*vnn(1,1)) + N[2]*(bdf0*vfrac(2,1) + bdf1*vn(2,1) + bdf2*vnn(2,1));
+const double crhs25 = crhs7*(DN(0,0)*vn(0,1) + DN(1,0)*vn(1,1) + DN(2,0)*vn(2,1));
+const double crhs26 = crhs9*(DN(0,1)*vn(0,1) + DN(1,1)*vn(1,1) + DN(2,1)*vn(2,1));
+const double crhs27 = crhs25 + crhs26;
+const double crhs28 = crhs12*(DN(0,0)*vfrac(0,1) + DN(1,0)*vfrac(1,1) + DN(2,0)*vfrac(2,1)) + crhs13*(DN(0,1)*vfrac(0,1) + DN(1,1)*vfrac(1,1) + DN(2,1)*vfrac(2,1));
+const double crhs29 = -crhs20*crhs4 - crhs21*crhs4 - crhs22*crhs4 + crhs24 - crhs25 - crhs26 + crhs28;
+const double crhs30 = N[1]*crhs4;
+const double crhs31 = N[1]*crhs17;
+const double crhs32 = crhs16*(DN(1,0)*crhs12 + DN(1,1)*crhs13);
+const double crhs33 = N[2]*crhs4;
+const double crhs34 = N[2]*crhs17;
+const double crhs35 = crhs16*(DN(2,0)*crhs12 + DN(2,1)*crhs13);
+rhs[0]=N[0]*crhs11 - N[0]*crhs14 - N[0]*crhs6 - crhs15*crhs18 - crhs15*crhs19 + crhs3*crhs5;
+rhs[1]=-N[0]*crhs24 + N[0]*crhs27 - N[0]*crhs28 - crhs18*crhs29 - crhs19*crhs29 + crhs23*crhs5;
+rhs[2]=N[1]*crhs11 - N[1]*crhs14 - N[1]*crhs6 - crhs15*crhs31 - crhs15*crhs32 + crhs3*crhs30;
+rhs[3]=-N[1]*crhs24 + N[1]*crhs27 - N[1]*crhs28 + crhs23*crhs30 - crhs29*crhs31 - crhs29*crhs32;
+rhs[4]=N[2]*crhs11 - N[2]*crhs14 - N[2]*crhs6 - crhs15*crhs34 - crhs15*crhs35 + crhs3*crhs33;
+rhs[5]=-N[2]*crhs24 + N[2]*crhs27 - N[2]*crhs28 + crhs23*crhs33 - crhs29*crhs34 - crhs29*crhs35;
+
+        //Add intermediate results to local system.
+        noalias(rRHS) += rhs * rData.Weight;
+    }
+
+    /***********************************************************************************/
+    /***********************************************************************************/
+    template <>
+    void VectorialConvectionFractionalElement<VectorialConvectionFractionalElementData<3, 4>>::ComputeGaussPointRHSContribution(VectorialConvectionFractionalElementData<3, 4> &rData,
+                                                                                                                                VectorType &rRHS)
+    {
+
+
+        const double h = rData.ElementSize;
+        const double dt = rData.DeltaTime;
+        const double bdf0 = rData.bdf0;
+        const double bdf1 = rData.bdf1;
+        const double bdf2 = rData.bdf2;
+        const double dyn_tau = rData.DynamicTau;
+        const auto &vn = rData.Velocity_OldStep1;
+        const auto &vnn = rData.Velocity_OldStep2;
+        // const auto &vnnn = rData.Velocity_OldStep3; #an_bdf2
+        const auto &vmesh = rData.MeshVelocity;
+        const auto &vfrac = rData.Velocity_Fractional;
+        const auto &vconv = vfrac - vmesh;
+
+        // Get shape function values
+        const auto &N = rData.N;
+        const auto &DN = rData.DN_DX;
+
+        // Stabilization parameters
+        constexpr double stab_c2 = 2.0;
```

Comment:
Instead of putting it in each function, I'd define these as `static constexpr` variables in the `private` section of the header. By doing this you have them defined only in one place, something that eases changing them if required.